src/js/util/dataset-utils.ts

```diff
@@ -59,7 +61,7 @@ function reset() {
  * @param {number} id - The ID of the dataset.
  * @returns {Array|string} An array of objects.
  */
-function input(id) {
+function input(id: number) {
```

Comment:
Good point! Yes, I think we should annotate the return values whenever possible. Will fix.

Reply:
I decided not to add a type annotation for the return value because I'm not yet sure why the jsdoc marked it as `Array | string`. Seems like an opportunity for a cleaner design. Something to come back to!